src/middlewares/createRouterMiddleware.ts

```diff
@@ -62,6 +62,11 @@ export const createRouterMiddleware = <
 
     const initialUiState = instantSearchInstance._initialUiState;
 
+    instantSearchInstance._initialUiState = {
+      ...initialUiState,
+      ...stateMapping.routeToState(router.read()),
+    };
```

Comment:
If ever we go for this solution, we need to document why this happens here.

But again, this doesn't seem too right to me. Perhaps we can manually call `subscribe()` where it's needed in VIS.

Reply:
That's an interesting idea